src/common/components/rangeDropdown/__tests__/RangeDropdown.test.tsx

```diff
@@ -0,0 +1,240 @@
+        name: /start integer/i,
+      })
+    ).toBeInTheDocument();
+
+    escKeyPressHelper();
+
+    // Assert that we can no longer find the menu content after we have pressed
+    // Escape.
+    expect(
+      screen.queryByRole('spinbutton', {
+        name: /start integer/i,
+      })
+    ).not.toBeInTheDocument();
+  });
+});
+
+test('should not open dropdown when user focuses toggle button', () => {
+  renderComponent();
+
+  const toggleButton = screen.getByRole('button', { name: title });
+  act(() => toggleButton.focus());
+
+  expect(
+    screen.queryByRole('spinbutton', {
+      name: /start integer/i,
+    })
+  ).not.toBeInTheDocument();
+});
+
+test('should open dropdown when user clicks on toggle button', () => {
+  renderComponent();
+
+  const toggleButton = screen.getByRole('button', { name: title });
+  userEvent.click(toggleButton);
+
+  expect(
+    screen.queryByRole('spinbutton', {
+      name: /start integer/i,
+    })
+  ).toBeInTheDocument();
+});
+
+describe('when dropdown has been closed, it should reopen with', () => {
+  const getClosedInput = async () => {
+    renderComponent();
+
+    const toggleButton = screen.getByRole('button', { name: title });
+    userEvent.click(toggleButton);
+
+    escKeyPressHelper();
+
+    expect(
+      screen.queryByRole('spinbutton', {
+        name: /start integer/i,
+      })
+    ).not.toBeInTheDocument();
+
+    expect(toggleButton).toHaveFocus();
+  };
+
+  test('ArrowDown', () => {
+    getClosedInput();
+
+    arrowDownKeyPressHelper();
+
+    expect(
+      screen.queryByRole('spinbutton', {
+        name: /start integer/i,
+      })
+    ).toBeInTheDocument();
+  });
+
+  test('ArrowUp', () => {
+    getClosedInput();
+
+    arrowDownKeyPressHelper();
+
+    expect(
+      screen.queryByRole('spinbutton', {
+        name: /start integer/i,
+      })
+    ).toBeInTheDocument();
+  });
+});
+
+test('should set default value', () => {
+  renderComponent();
+
+  const toggleButton = screen.getByRole('button', { name: title });
+  userEvent.click(toggleButton);
+
+  const minValueTextbox = screen.queryByRole('spinbutton', {
+    name: /start integer/i,
+  }) as HTMLInputElement;
+  const maxValueTextbox = screen.queryByRole('spinbutton', {
+    name: /end integer/i,
+  }) as HTMLInputElement;
+  const setDefaultsCheckbox = screen.queryByRole('checkbox', {
+    name: /set default values/i,
+  });
+
+  expect(minValueTextbox).toBeInTheDocument();
+  expect(minValueTextbox.value).toBe('0');
+  expect(maxValueTextbox).toBeInTheDocument();
+  expect(maxValueTextbox.value).toBe('100');
+  expect(setDefaultsCheckbox).toBeInTheDocument();
+
+  //click defaults
+  userEvent.click(setDefaultsCheckbox);
+
+  expect(minValueTextbox).toHaveAttribute('disabled');
+  expect(maxValueTextbox).toHaveAttribute('disabled');
+
+  //todo: for some reason value not changing with disabled attribute
+  //expect(minValueTextbox.value).toBe('18');
+  //expect(maxValueTextbox.value).toBe('80');
```

Comment:
I think this doesn't work because the fixed values are provided as basic values props when checkbox is checked

Reply:
to test this you need to use the rerender that renderComponent func returns